chef/cookbooks/cinder/recipes/ceph.rb

```diff
@@ -37,9 +37,17 @@
   if ceph_servers.length > 0
     include_recipe "ceph::keyring"
   else
-    message = "Ceph was not deployed with Crowbar yet!"
-    Chef::Log.fatal(message)
-    raise message
+    # If we don't have any osd from our query, it could be because the
+    # osd nodes temporarily lost the role (while rebooting, for instance).
+    # So check if the ceph setup was done once already, to decide whether
+    # to fail or just to emit a warning.
+    if File.exists?("/etc/ceph/ceph.client.admin.keyring")
```

Comment:
In the case of "deployed by crowbar" ceph cluster, we always use this hard-coded value (and ignore the attribute, since it's not known by the ceph recipes). Since this block is only about this case, that's why I used the hard-coded value.